Sources/SwiftPlot/Renderer.swift

```diff
@@ -194,3 +194,35 @@ extension Renderer {
         (self.xOffset, self.yOffset) = oldOffset
     }
 }
+
+/// Polygon structure definition
+public struct Polygon {
+    public var points: [Point] {
+        didSet {
+            let count = points.count
+            precondition(count >= 2, "Polygon: points array should always contain at least 3 points, it now has \(count).")
+        }
+    }
+
+    public init?(_ points: [Point]) {
```

Comment:
It might also be nice to have a version which takes a variadic list of Points, so instead of:

```
Polygon([tL, tR, bR, bL])!
```

we could write

```
Polygon(tL, tR, bR, bL)!
```

without the square brackets.

Reply:
I like that, didn't think about using variadics!